File: 1545-find-kth-bit-in-nth-binary-string/1545-find-kth-bit-in-nth-binary-string.cpp
class Solution {
public:
    string S(int i){
        if(i==1)    return "0";
        string temp=S(i-1)+"1";
        for(int i=temp.size()-2;i>-1;i--)  temp+=temp[i]=='0'?'1':'0';
        return temp;
    }
    
    char findKthBit(int n, int k) {
        return S(n)[k-1];
    }
};

File: 1545-find-kth-bit-in-nth-binary-string/1545-find-kth-bit-in-nth-binary-string.py
class Solution:
    def findBinaryString(self, n):
        if n == 1:
            return '0'
        binarySub = self.findBinaryString(n-1)
        inverted = binarySub.replace('1', '2').replace('0', '1').replace('2', '0')

        return binarySub+'1'+inverted[::-1]
    
    def findKthBit(self, n: int, k: int) -> str:
        return self.findBinaryString(n)[k-1]